.github/actions/secure-project-checkout/action.yaml

```
@@ -33,20 +38,27 @@ runs:
       shell: bash
       env:
         CONTEXT: "${{ toJSON(github) }}"
+        INPUT_SHA1: "${{ inputs.sha1 }}"
       working-directory: "${{ inputs.path }}"
       run: |
         set -euo pipefail
 
         git_sha="$(git log -1 --format='%H')"
         github_sha="$GITHUB_SHA"
+        expected_sha1="$github_sha"
```

You don't really need the `github_sha` anymore do you?
```suggestion
        expected_sha1="$GITHUB_SHA"
```